### PR TITLE
ci: Update hedera-local-node for CI best practices

### LINKED ISSUE #591: update ci workflows for security & best practices
we need to update the Github Actions CI pipelines to follow industry and organization best practices. This includes but is not limited to using commit locked versions of third-party actions, using self-hosted runners with advanced security, and adding explicit permissions to each workflow.

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:

This PR is intended to provide security and CI best practice updates to the github actions workflows. 

This PR modifies files in the .github folder in order to support best practice updates

**Related issue(s)**:

Fixes #591 

**Notes for reviewer**:

Actions that don't run as part of a pull request are manually tested with a workflow-dispatch trigger prior to PR readiness.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/CODEOWNERS
Comment (georgi-l95) [inline .github/CODEOWNERS:1]: Make codeowners for all files in this repo @georgi-l95(me), @isavov, @Ivo-Yankov, @natanasow, @konstantinabl and @Ivo-Yankov .

File: .github/dependabot.yml
Comment (georgi-l95) [inline .github/dependabot.yml:31]: add newline at end of file

File: .github/workflows/flow-pull-request-checks.yaml
Comment (georgi-l95) [inline .github/workflows/flow-pull-request-checks.yaml:144]: add newline at end of file

**Review by georgi-l95 — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (rbarkerSL) [reply .github/CODEOWNERS:1]: Modified.

Comment (georgi-l95) [reply .github/CODEOWNERS:1]: Thank you 🚀

**Review by georgi-l95 — APPROVED** [review]
Unblocking for codeowners file changes. Don't have much context for the other changes, they look good to me, but someone from devops should take a look as well 👍

**Pull request merged**
